PROTOCOL.md

```diff
@@ -881,7 +881,16 @@ Field Name | Description
 name| Name of this (possibly nested) column
 type| String containing the name of a primitive type, a struct definition, an array definition or a map definition
 nullable| Boolean denoting whether this field can be null
-metadata| A JSON map containing information about this column. Keys prefixed with `Delta` are reserved for the implementation. See [TODO](#) for more information on column level metadata that clients must handle when writing to a table.
+metadata| A JSON map containing information about this column. Keys prefixed with `Delta` are reserved for the implementation. See [metadata](#metadata) for more information on column level metadata that clients must handle when writing to a table.
+
+### metadata
+This field include column mapping information to avoid any column naming restrictions.
+
+Field Name | Description
```

Comment:
You can give an example of what types of keys/values are stored in the metadata. which is just a map of String->JSON-encoded value.

For example, the Column Mapping section gives some examples. he physical name is stored as part of the column metadata with the key delta.columnMapping.physicalName. The column id is stored within the metadata with the key delta.columnMapping.id. The column mapping is governed by the table property delta.columnMapping.mode and can be one of none, id, and name.
```

Generated columns also gives an example. The metadata for a column in the table schema MAY contain the key delta.generationExpression.
```
The metadata for a column in the table schema MAY contain the key delta.generationExpression.
```
The physical name is stored as part of the column metadata with the key delta.columnMapping.physicalName. The column id is stored within the metadata with the key delta.columnMapping.id. The column mapping is governed by the table property delta.columnMapping.mode and can be one of none, id, and name.
```